src/NovelRT/DotNet/RuntimeService.cpp

```diff
@@ -115,7 +115,7 @@ namespace NovelRT::DotNet {
       if (result != 0)
       {
         _logger.logError("Failed to initialize the runtime: {}", result);
-        throw std::runtime_error("Failed to initialize the runtime");
+        throw Exceptions::InitialisationFailureException("Failed to initialize the runtime");
```

Comment:
Again, we might want to consider somehow making InitialisationFailureException a more standardised error and then we can provide the inner reason of runtime failure ourselves for a better UX?